src/main/kotlin/org/malachite/estella/task/api/TaskController.kt

```diff
@@ -32,28 +32,36 @@ class TaskController(
             @RequestParam("owner") organizationUuid: String?,
             @RequestParam("taskStage") taskStageUuid: String?,
             @RequestParam("devMail") devMail: String?,
-            @RequestHeader(EStellaHeaders.devPassword) password: String
+            @RequestParam("interview") interviewUuid: String?,
+            @RequestHeader(EStellaHeaders.devPassword) password: String?
     ): ResponseEntity<Any> {
-        if (!areParamsValid(organizationUuid, taskStageUuid, devMail))
-            return ResponseEntity.badRequest().body(Message("Exactly one of parameters: organizationUuid, taskStageUuid and devMail is required"))
+        if (!areParamsValid(organizationUuid, taskStageUuid, devMail, interviewUuid))
+            return ResponseEntity.badRequest().body(Message("Exactly one of parameters: organizationUuid, taskStageUuid, devMail is required"))
+        if (!isPasswordProvided(organizationUuid, taskStageUuid, devMail, password) && interviewUuid == null)
+            return OwnResponses.UNAUTH
         val tasks: List<TaskDto> = organizationUuid
                 ?.let {
-                    devMail?.let { taskStageService.getTasksByDev(it, password) }
-                            ?: taskStageService.assertDevPasswordCorrect(organizationUuid, password)
+                    devMail?.let { taskStageService.getTasksByDev(it, password!!) }
```

Comment:
Naprawdę bym tu proponował tego whena, jak jest w notatkach, ale to już szczegół teraz

Reply:
don